extensions/sockets/src/main/java/org/mule/module/socket/api/client/TcpRequesterClient.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.socket.api.client;
+
+import static org.mule.module.socket.internal.NetworkUtils.getSocketAddress;
+import org.mule.module.socket.api.exceptions.UnresolvableHostException;
+import org.mule.module.socket.api.protocol.TcpProtocol;
+import org.mule.module.socket.api.tcp.TcpClientSocketProperties;
+import org.mule.runtime.api.connection.ConnectionException;
+
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+import java.net.SocketException;
+
+public class TcpRequesterClient extends AbstractSocketClient implements RequesterSocket
+{
+
+    private Socket socket;
+    private TcpClientSocketProperties socketProperties;
+    private TcpProtocol protocol;
+
+    public TcpRequesterClient(TcpClientSocketProperties socketProperties, TcpProtocol protocol, String host, Integer port) throws ConnectionException
+    {
+        super(host, port);
+        this.socket = new Socket();
+        this.protocol = protocol;
+        this.socketProperties = socketProperties;
+        configureSocket();
+    }
+
+    private void configureSocket() throws ConnectionException
+    {
+        try
+        {
+            if (socketProperties.getKeepAlive() != null)
+            {
+                socket.setKeepAlive(socketProperties.getKeepAlive());
+            }
+
+            if (socketProperties.getSendBufferSize() != null)
+            {
+                socket.setSendBufferSize(socketProperties.getSendBufferSize());
+            }
+
+            if (socketProperties.getReceiveBufferSize() != null)
+            {
+                socket.setReceiveBufferSize(socketProperties.getReceiveBufferSize());
+            }
+
+            if (socketProperties.getLinger() != null)
+            {
+                socket.setSoLinger(true, socketProperties.getLinger());
+            }
+
+
+            socket.setTcpNoDelay(socketProperties.getSendTcpNoDelay());
+            socket.setSoTimeout(socketProperties.getTimeout());
+            socket.bind(null);
```

Comment:
exactly. Shouldn't the user have the chance to specify a port if he wants? It should be optional and default to null